File: artisanat-backend/src/main/java/com/artisanat_backend/dto/response/SubOrderResponseDto.java
package com.artisanat_backend.dto.response;

import com.artisanat_backend.model.Artisan;
import com.artisanat_backend.model.Product;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SubOrderResponseDto {
    private Long id;
    private double subTotal;
    private Artisan artisan;
    private List<SubOrderItemResponseDto> subOrderItems;
}

File: artisanat-frontend/src/app/core/dtos/response/sub-order-response-dto.ts
import {Artisan} from "../../models/artisan.model";
import {SubOrderItemResponseDto} from "./sub-order-item-response-dto";

export interface SubOrderResponseDto {
  id: number;
  subTotal: number;
  artisan: Artisan;
  subOrderItems: SubOrderItemResponseDto[];
}
